sdk/java/pulumi/src/main/java/com/pulumi/serialization/internal/Converter.java

```diff
@@ -437,7 +437,8 @@ private Object tryConvertObjectInner(
                             argument == null ? "null" : argument.getClass()
                     ), e);
                 } catch (IllegalAccessException | InvocationTargetException e) {
-                    throw new IllegalStateException(String.format("Unexpected exception: %s", e.getMessage()), e);
+                    var exMsg = e.getMessage() == null ? e.getClass().getSimpleName() : e.getMessage();
```

Comment:
This went in through another PR I think so the diff will disappear after we rebase on main.